File: be/models/RepairTicket.js
const { DataTypes } = require('sequelize');
const { sequelize } = require('../config/database');

const RepairTicket = sequelize.define('RepairTicket', {
  id: {
    type: DataTypes.INTEGER,
    primaryKey: true,
    autoIncrement: true
  },
  vehicleId: {
    type: DataTypes.INTEGER,
    allowNull: false,
    references: {
      model: 'vehicles',
      key: 'id'
    }
  },
  totalAmount: {
    type: DataTypes.DECIMAL(10, 2),
    allowNull: false,
    defaultValue: 0
  },
  mechanicName: {
    type: DataTypes.STRING,
    allowNull: true,
    defaultValue: 'Chưa phân công'
  },
  status: {
    type: DataTypes.ENUM('draft', 'working', 'completed', 'paid'),
    allowNull: false,
    defaultValue: 'draft'
  },
  startedAt: {
    type: DataTypes.DATE,
    allowNull: true,
    comment: 'Timestamp when repair started (status changed to working)'
  },
  completedAt: {
    type: DataTypes.DATE,
    allowNull: true,
    comment: 'Timestamp when repair completed (status changed to completed)'
  },
  paidAt: {
    type: DataTypes.DATE,
    allowNull: true,
    comment: 'Timestamp when payment received (status changed to paid)'
  }
}, {
  tableName: 'repair_tickets',
  timestamps: true
});

module.exports = RepairTicket;

File: be_python/models/repair_ticket.py
from sqlalchemy import Column, Integer, String, Numeric, DateTime, ForeignKey
from sqlalchemy.orm import relationship
from sqlalchemy.sql import func

from database.engine import Base


class RepairTicket(Base):
    """
    Repair ticket (phiếu sửa chữa) linking a vehicle to a list of repair items.
    Tracks status transitions: draft → working → completed → paid.
    """

    __tablename__ = "repair_tickets"

    id = Column(Integer, primary_key=True, autoincrement=True)
    vehicleId = Column(
        "vehicleId", Integer, ForeignKey("vehicles.id"), nullable=False, index=True
    )
    totalAmount = Column("totalAmount", Numeric(10, 2), nullable=False, default=0)
    mechanicName = Column("mechanicName", String, nullable=True, default="Chưa phân công")
    status = Column(String, nullable=False, default="draft")  # draft|working|completed|paid
    startedAt = Column("startedAt", DateTime, nullable=True)
    completedAt = Column("completedAt", DateTime, nullable=True)
    paidAt = Column("paidAt", DateTime, nullable=True)
    createdAt = Column("createdAt", DateTime, server_default=func.now())
    updatedAt = Column("updatedAt", DateTime, server_default=func.now(), onupdate=func.now())

    # Relationships
    vehicle = relationship("Vehicle", back_populates="repairTickets", lazy="selectin")
    items = relationship(
        "RepairItem", back_populates="ticket", lazy="selectin", cascade="all, delete-orphan"
    )
